www/js/overlay.js

```diff
@@ -176,6 +176,46 @@ const resizeButtonFactory = (increaseFunc, decreaseFunc) =>
     ),
   );
 
+const closeAllDropDowns = () => {
+  document
+    .querySelectorAll('.options-container.visible')
+    .forEach(el => el.classList.remove('visible'));
+};
+
+const toggleDropDown = $select => {
+  closeAllDropDowns();
+  const $options = $select.querySelector('.options-container');
+  $options.classList.add('visible');
+  $options.style.top = `${
+    window.innerHeight < $options.offsetHeight + $select.offsetTop
+      ? -$options.offsetHeight
+      : $select.querySelector('.select-value').offsetHeight
+  }px`;
+};
+
+window.addEventListener('click', e => {
+  if (!e.target.classList.contains('select-value')) {
+    closeAllDropDowns();
+  } else {
+    toggleDropDown(e.target.parentElement);
+  }
+});
+
+const fontListFactory = (list, propName) => {
+  const lowerize = str => str.replace(/ /g, '').toLowerCase();
+  const onClickOption = (e, font) => {
+    e.target.parentElement.parentElement.querySelector('.select-value').innerHTML = font;
+    overlayVars[propName] = lowerize(font);
+    savePrefs();
+  };
+  const options = list.map(font => h(`div.option`, { onclick: e => onClickOption(e, font) }, font));
+  return h(
+    'div.custom-select',
```

Comment:
why didn't we just use a select element like we have in other parts of the app?

Reply:
It wasn't looking as good here - and also needed some functionality which allows me to open the dropdown as drop-up depending on window size.